tests/trainer/test_dataloaders.py

```diff
@@ -311,37 +302,32 @@ def test_dataloaders_with_limit_train_batches(tmpdir, dataset, limit_train_batch
 
     trainer.fit(model, train_dataloaders=train_dl, val_dataloaders=val_dl)
     assert trainer.state.finished, f"Training failed with {trainer.state}"
-    assert trainer.num_training_batches == (limit_train_batches if limit_train_batches != 0.0 else float("inf"))
-    assert epoch_cb.train_epoch_count == (epochs if limit_train_batches > 0 else 0)
-    assert epoch_cb.train_batches_seen == limit_train_batches * epochs
+    assert trainer.num_training_batches == limit_train_batches
+    assert epoch_cb.train_epoch_count == max_epochs
+    assert epoch_cb.train_batches_seen == limit_train_batches * max_epochs
 
 
 @pytest.mark.parametrize(
-    ["dataset", "limit_val_batches"],
+    "dataset",
     [
-        (RandomDataset(32, 128), 0),
-        (RandomDataset(32, 128), 10),
-        (RandomIterableDataset(32, 128), 0),
-        (RandomIterableDataset(32, 128), 10),
-        (RandomIterableDatasetWithLen(32, 128), 0),
-        (RandomIterableDatasetWithLen(32, 128), 10),
+        RandomDataset(32, 128),
+        RandomIterableDataset(32, 128),
+        RandomIterableDatasetWithLen(32, 128),
     ],
 )
-def test_dataloaders_with_limit_val_batches(tmpdir, dataset, limit_val_batches):
+def test_dataloaders_with_limit_val_batches(tmpdir, dataset):
     """Verify inf train, val & test dataloaders (e.g. IterableDataset) passed with batch limit as number."""
 
     epoch_cb = Counter()
     callbacks = [epoch_cb]
     enable_checkpointing = False
-    if limit_val_batches > 0:
-        callbacks.append(ModelCheckpoint(monitor="val_log", save_top_k=1, mode="max", verbose=False))
-        enable_checkpointing = True
```

Comment:
since you are removing the case where limit_val_batches == 0, this if statement would have to stay.